R/calculate.R

```diff
@@ -152,3 +162,39 @@ calculate_exposure_by_technology_and_company <- function(asset_type,
 
   return(exposure_by_technology_and_company_filtered)
 }
+
+calculate_terminal_value <- function(data,
+                                     end_year,
+                                     growth_rate,
+                                     discount_rate) {
+  terminal_value <- data %>%
+    dplyr::filter(.data$year == .env$end_year) %>%
+    dplyr::group_by(
+      .data$investor_name, .data$portfolio_name, .data$id,
```

Comment:
please reconsider if really all grouping vars (e.g. phase_out) are needed

Reply:
shouldnt that be resolvable by using mutate instead of summarise? 🤔

Reply:
yes, but then we would keep the production values from the end_year, which does not make sense from a model point of view. So I could do that, but then I would explicitly need to set some NA values here for the production fields

Reply:
done

Reply:
turns out summarizing leads to issues further down the line because the financial values were missing for the terminal value row. I ended up changing it to a mutate after all.